app/utils/filters.py

```diff
@@ -106,3 +106,30 @@ def compare_lists(source, target, mode="sync"):
         filtered_target_groups.sort(key=lambda x: x[target_key])
 
         return filtered_source_groups, filtered_target_groups
+
+
+def get_unique_nested_dicts(source_items, nested_key):
+    """Get the unique items from a list located in a dict or from a list of dicts with the same data schema.
+    Considers the whole object for uniqueness, not specific keys.
+
+    Args:
+        groups (list or dict): A list of groups or a single group.
+        key (str): The key to get the users from the groups.
```

Comment:
Minor comment that the method docstring args need to be updated.

Reply:
Thanks for spotting!!